docs/api.rst

```diff
@@ -299,11 +299,115 @@ Given a higher-order stream, return a new stream that merges all the inner strea
   stream:        -s------t--------->
   join(stream):  ---a---b--4c-5-d6->
 
-concatMap :: (a -> Stream b) -> Stream a -> Stream b
+.. _concatMap:
 
-mergeConcurrently :: int -> Stream (Stream a) -> Stream a
+concatMap
+^^^^^^^^^
+
+.. code-block:: haskell
+
+  concatMap :: (a -> Stream b) -> Stream a -> Stream b
+
+Transform each event in stream into a stream, and then concatenate it onto the 
+end of the resulting stream. Note that f must return a stream.
+
+The mapping function f is applied lazily. That is, f is called only once it is 
+time to concatenate a new stream.
+
+``function f(x) -> Stream``
+
+.. code-block:: javascript
+
+  stream:                -a----b----c|
+  f(a):                   1--2--3|
+  f(b):                        1----2----3|
+  f(c):                               1-2-3|
+  concatMap(f, stream):  -1--2--31----2----31-2-3|
+  f called lazily:        ^      ^          ^
+
+Note the difference between concatMap and chain: concatMap concatenates, while 
+chain merges.
+
+.. _mergeConcurrently:
+
+mergeConcurrently
+^^^^^^^^^^^^^^^^^
+
+.. code-block:: haskell
+
+  mergeConcurrently :: int -> Stream (Stream a) -> Stream a
+
+Given a higher-order stream, return a new stream that merges inner streams as 
+they arrive up to the specified concurrency. Once concurrency number of streams 
+are being merged, newly arriving streams will be merged after an existing one 
+ends.
+
+.. code-block:: javascript
```

Comment:
Same here re: ending preceding paragraph with `::` and removing `.. code-block:: javascript`